experimental/segmenter/src/provider.rs

```diff
@@ -132,7 +132,7 @@ pub struct LstmMatrix<'data> {
 }
 
 #[cfg(feature = "lstm")]
-impl<'data> LstmMatrix<'data> {
+impl<'data> ZeroMatrix32<'data> {
```

Comment:
nit: `ZeroMatrix32` and `MatrixZero<D>` are confusing names. Maybe we should unify them...

Reply:
If we are going to unify them then no, otherwise I'd like better names.

Reply:
Well your PR touches them, too, and there's a thread over there where you made a case for them to stay separate

Reply:
I do like having them separate. I could see them having the *same* name, just as `provider::ZeroMatrix` and `math::ZeroMatrix`. That'd be less confusing than trying to figure out what the difference between a `ZeroMatrix` and a `MatrixZero` is.

Reply:
I don't see changing the name of the internal (non provider) matrix types as in scope for this PR

Reply:
Well you're producing a name collision, so I don't see why not. The names have only been there for like 2 days.

Reply:
There is

VarZeroVecOwned
Cow::Owned
ZeroVec::Owned
serde::DeserializeOwned

but "Zero" usually goes at the beginning in ZeroVec, ZeroMap, ...

so how about,

- math_helper::Matrix
- math_helper::MatrixOwned
- math_helper::MatrixBorrowed
- math_helper::ZeroMatrixBorrowed
- provider::ZeroMatrix32

Reply:
sg